core/roboconf-core/src/main/java/net/roboconf/core/model/helpers/ComponentHelpers.java

```diff
@@ -250,12 +266,12 @@ public static String findComponentInstaller( Component component ) {
 	/**
 	 * Finds all the imported variables for a given component.
 	 * @param component a non-null component
-	 * @return a non-null map of imported variables (key = variable name, value = true if optional, false if required).
+	 * @return a non-null map of imported variables (key = variable name, value = imported variable bean)
 	 */
-	public static Map<String,Boolean> findAllImportedVariables( Component component ) {
+	public static Map<String,ImportedVariable> findAllImportedVariables( Component component ) {
 
 		// Process components from the ancestors to the children... => override
-		Map<String,Boolean> result = new HashMap<String,Boolean> ();
+		Map<String,ImportedVariable> result = new HashMap<> ();
```

Comment:
With Java 7, you do not have to write generics anymore in implementations.
They are deduced by the compiler from the **Map** declaration. :wink: